### PR TITLE
Implement [HomeTabNavigator]

### PR DESCRIPTION
**Pull request by jb9229**
## Description

I implemented 'HomeTabNavigator'.
I add empty screens.
- Home
- HomeCalendar
- HomeFavorite
- HomeSetting

The navigator is changed color when it is active
  - i don't use theme color, i fixed the static variable in /src/components/shared/TabBarIcon.tsx
  - i referenced the icon color in the IssueHunt picture.
  - i use the icon [expo vector icon](https://expo.github.io/vector-icons/)
  ※ COLOR_TABICON_DEFAULT = '#c6cbcf';
  ※ COLOR_TABICON_SELECTED = '#75a3ff';

## Related Issues

Implement [HomeTabNavigator] navigation #57 

## Tests

I added the following tests:
  1. Home.test.tsx, HomeCalendar.test.tsx, HomeFavorite.test.tsx, HomeSetting.test.tsx
    - it is just empty screen for navigator so default test(test file make by dooboo-cli screen)
  2. TabBarIcon.test.tsx
    - I add UI render test(snapshot test)
    - I tried to do a color change test but it failed unfortunately, I will add it after knowing the future technology.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide](https://github.com/dooboolab/whatssub/blob/master/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] Run `yarn test` or `yarn test -u` if you need to update snapshot.
- [x] Run `yarn lint`
- [x] I am willing to follow-up on review comments in a timely manner.

### CODE REVIEW

File: src/components/shared/TabBarIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import * as Icon from '@expo/vector-icons';
+import React from 'react';
+import { StyleSheet } from 'react-native';
+
+// Static Variable
+const DEFAULT_ICON_SIZE = 32;
+const COLOR_TABICON_DEFAULT = '#c6cbcf';
+const COLOR_TABICON_SELECTED = '#75a3ff';
+
+const styles = StyleSheet.create({
```
Comment (hyochan) [inline src/components/shared/TabBarIcon.tsx]: Please do not use `StyleSheet` and rather use `styled-components`. We then need to add `typings` on `StyleSheet` which may make code bit messy.

File: src/components/navigation/HomeTabNavigator.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+});
+
+HomeSettingStack.navigationOptions = {
+  tabBarIcon: SettingIcon,
+};
+
+/**
+ * Setting Route Configuration
+ */
+const styles = StyleSheet.create({
```
Comment (hyochan) [inline src/components/navigation/HomeTabNavigator.tsx]: Same here. Please don't use `StyleSheet`.

File: src/components/shared/TabBarIcon.tsx (hunk 1 above)
Comment (jb9229) [reply src/components/shared/TabBarIcon.tsx]: @hyochan I replace StyleSheet with styled-components

File: src/components/navigation/HomeTabNavigator.tsx (hunk 2 above)
Comment (jb9229) [reply src/components/navigation/HomeTabNavigator.tsx]: @hyochan I replace StyleSheet with styled-components

**Comment by hyochan** [comment]
@all-contributors please add jb9229 for code, test

**Pull request merged**